scripts/TunnelAzure.ps1

```diff
@@ -66,96 +66,87 @@ Function Remove-ResourceGroup {
+        $Context.TunnelServiceIPv6Address = $Context.TunnelServiceIPv6Address.publicIp
+        az network nic create --resource-group $Context.ResourceGroup --name $nicName --vnet-name $Context.VnetName --subnet $Context.SubnetName --public-ip-address $publicIPv4Name --network-security-group $Context.NSGName | Out-Null 
+        az network nic ip-config create --resource-group $Context.ResourceGroup --name $IPv6ConfigName --nic-name $nicName --private-ip-address-version IPv6 --vnet-name $Context.VnetName --subnet $Context.SubnetName --public-ip-address $publicIPv6Name | Out-Null  
+        $script:Context.ServiceNicName = $nicName    
+    }
 }
 
 Function Remove-SSHRule {
     if (!$Context.WithSSHOpen) {
         Write-Header "Removing SSH rule..."
-        az network nsg rule delete --resource-group $Context.ResourceGroup --nsg-name "$($script:Context.VnetName)-NSG" -n "AllowSSHIN" --only-show-errors | Out-Null
+        az network nsg rule delete --resource-group $Context.ResourceGroup --nsg-name $Context.NSGName -n "AllowSSHIN" --only-show-errors | Out-Null
     }
 }
 
 Function New-TunnelVM {    
     Write-Header "Creating VM '$($Context.VmName)'..."
-    az vm create --location $Context.Location --resource-group $Context.ResourceGroup --name $Context.VmName --image $Context.Image --size $Context.Size --ssh-key-values "$($Context.SSHKeyPath).pub" --public-ip-address-dns-name $Context.VmName --admin-username $Context.Username --vnet-name $Context.VnetName --subnet $Context.SubnetName --only-show-errors | Out-Null
+    az vm create --location $Context.Location --resource-group $Context.ResourceGroup --name $Context.VmName --image $Context.Image --size $Context.Size --ssh-key-values "$($Context.SSHKeyPath).pub" --public-ip-address-dns-name $Context.VmName --admin-username $Context.Username --nics $Context.TunnelNicName --only-show-errors | Out-Null
 
     if ($Context.BootDiagnostics) {
         Write-Header "Enabling boot diagnostics..."
         az vm boot-diagnostics enable --resource-group $Context.ResourceGroup --name $Context.VmName
     }
 }
 
-Function New-NetworkRules {
-    Write-Header "Creating network rules..."
-    
-    if ($Context.WithSSHOpen) {
-        az network nsg rule create --resource-group $Context.ResourceGroup --nsg-name "$($Context.VmName)NSG" -n SSHIN --priority 100 --source-address-prefixes '*' --source-port-ranges '*' --destination-address-prefixes '*' --destination-port-ranges 22 --access Allow --protocol Tcp --description "Allow SSH" > $null
-    }
-    
-    az network nsg rule create --resource-group $Context.ResourceGroup --nsg-name "$($Context.VmName)NSG" -n HTTPIN --priority 101 --source-address-prefixes 'Internet' --source-port-ranges '*' --destination-address-prefixes '*' --destination-port-ranges 443 --access Allow --protocol '*' --description "Allow HTTP" > $null
-}
-
-Function New-AdvancedNetworkRules {
```

Comment:
Have we tested these changes in a personal subscription? I think these rules were needed for personal subs.